File: fyc-api-rest/src/main/java/fr/esgi/fyc_api_rest/rest/SeanceRestController.java
package fr.esgi.fyc_api_rest.rest;

import fr.esgi.fyc_api_rest.business.Seance;
import fr.esgi.fyc_api_rest.dto.request.SeanceRequestDTO;
import fr.esgi.fyc_api_rest.dto.response.SeanceResponseDTO;
import fr.esgi.fyc_api_rest.mapper.SeanceMapper;
import fr.esgi.fyc_api_rest.service.SeanceService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.stream.Collectors;

@RestController
@RequestMapping("/api/Seances")
public class SeanceRestController {
    private final SeanceMapper seanceMapper;
    private final SeanceService seanceService;

    public SeanceRestController(SeanceMapper seanceMapper, SeanceService seanceService) {
        this.seanceMapper = seanceMapper;
        this.seanceService = seanceService;
    }

    @PostMapping
    @ResponseStatus(code= HttpStatus.CREATED)
    public SeanceResponseDTO create(@RequestBody SeanceRequestDTO seanceDTO){
        Seance seance = seanceService.create(seanceMapper.toEntity(seanceDTO));
        return seanceMapper.toResponseDTO(seance);
    }

    @GetMapping
    @ResponseStatus(code = HttpStatus.OK)
    public List<SeanceResponseDTO> findAll(){
        List<Seance> seances = seanceService.findAll();
        return seances.stream().map(seanceMapper::toResponseDTO).toList();
    }

    @GetMapping("/{id}")
    @ResponseStatus(code = HttpStatus.OK)
    public SeanceResponseDTO findById(@PathVariable Long id){
        Seance seance = seanceService.findById(id);
        return seanceMapper.toResponseDTO(seance);
    }

    @PutMapping("/{id}")
    @ResponseStatus(code = HttpStatus.OK)
    public SeanceResponseDTO update(@PathVariable Long id, @RequestBody SeanceRequestDTO seanceDTO){
        Seance seance = seanceService.update(id, seanceMapper.toEntity(seanceDTO));
        return seanceMapper.toResponseDTO(seance);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(code = HttpStatus.NO_CONTENT)
    public void delete(@PathVariable Long id){
        seanceService.delete(id);
    }
}

File: fyc-rest-api-csharp-test/CometeAPI/Rest/SeanceRestController.cs
using FycRestApi.Business;
using FycRestApi.DTO.Request;
using FycRestApi.DTO.Response;
using FycRestApi.Mapper;
using FycRestApi.Service.Interface;
using Microsoft.AspNetCore.Mvc;

namespace FycRestApi.Rest;

[ApiController]
[Route("api/Seances")]
public class SeanceRestController : ControllerBase
{
    private readonly ISeanceService _seanceService;
    private readonly SeanceMapper _seanceMapper;
    public SeanceRestController(ISeanceService seanceService, SeanceMapper seanceMapper)
    {
        this._seanceService = seanceService;
        this._seanceMapper = seanceMapper;
    }

    [HttpGet("")]
    public async Task<ActionResult<List<SeanceResponseDTO>>> index()
    {
        List<Seance> seances = await this._seanceService.findAll();
        return Ok(seances.Select(seance => _seanceMapper.toDTO(seance)).ToList());
    }

    [HttpPost]
    public async Task<ActionResult<SeanceResponseDTO>> Create([FromBody] SeanceRequestDTO requestDTO)
    {
        try
        {
            Seance seance = await _seanceService.save(await _seanceMapper.toEntity(requestDTO));
            string uri = Url.Action(action: nameof(FindById), new { id = seance.id });
            return Created(uri, seance);
        }
        catch (Exception ex)
        {
            return BadRequest(ex.Message);
        }
    }

    [HttpGet("{id}")]
    public async Task<ActionResult<SeanceResponseDTO>> FindById(int id)
    {
        try
        {
            Seance seance = await _seanceService.findById(id);
            return Ok(_seanceMapper.toDTO(seance));
        }
        catch (KeyNotFoundException ex)
        {
            return NotFound(ex.Message);
        }
        catch (Exception ex)
        {
            return BadRequest(ex.Message);
        }
    }

    [HttpPut("{id}")]
    public async Task<ActionResult<SeanceResponseDTO>> update(int id, [FromBody] SeanceRequestDTO requestDTO)
    {
        try
        {
            Seance seance = await _seanceService.update(id, await _seanceMapper.toEntity(requestDTO));
            return Ok(_seanceMapper.toDTO(seance));
        }
        catch (KeyNotFoundException ex)
        {
            return NotFound(ex.Message);
        }
        catch (Exception ex)
        {
            return BadRequest(ex.Message);
        }
    }

    [HttpDelete("{id}")]
    public async Task<IActionResult> delete(int id)
    {
        try
        {
            await _seanceService.delete(id);
            return NoContent();
        }
        catch (KeyNotFoundException ex)
        {
            return NotFound(ex.Message);
        }
        catch (Exception ex)
        {
            return BadRequest(ex.Message);
        }
    }
}